### DISCUSSION TITLE
Allgemeine Frage zu den beiden Versionen

### DISCUSSION THREAD

**Discussion by woupi** [camueller/SmartApplianceEnabler#349] [Allgemein]
Guten Morgen Axel,

es gibt ja nun schon länger 2 Versionen (1.6.x und 2.0.x)
Eventuell hast Du es irgendwo geschrieben und ich habe es einfach nur übersehen.
Ich stell mir immer wider die selben Fragen und habe mich nun entschlossen diese mal zu stellen:
Ich habe die 1.6.x installiert, sollte ich auf die 2.0.x umstellen? (Vorteile/Nachteile)
Was sind die Unterschiede?
Warum gibt es noch beide? (Ist ja doppelt soviel Arbeit für dich, ergo muss es ja einen Triftigen Grund geben)
Warum ist die 2.0.x noch im Beta Stadium bzw. soll das so bleiben? (Auch dafür müsste es ja Gründe geben)

Vielen Dank für die Info
LG

**Comment by camueller** [comment]
Die 2.0 ist eigentlich nicht mehr "Beta", aber ich kann sie erst zur "offiziellen" Version machen, wenn das Upgrade von 1.6 auch für jemanden ohne Linux-Kenntniss möglich ist. Dazu muss ich noch ein Script erstellen, die Dokumenation anpassen und testen, dass wirklich alles reibungslos läuft. Diese Aktivitäten sind für die nächsten Wochen geplant. Wenn das alles so klappt, wird es keine weitere 1.6-Version geben. Ich selbst nutze auf beiden produktiven Raspis die 2.0-Version seit etlichen Monaten.

Vorteile der 2.0-Version:
- MQTT-basiert für bessere Integration in Hausautomatisierungen
- detaillierteres, konfigurierbares Dashboard auf Basis von NodeRED (da habe ich leider bisher wenig Feedback erhalten)
- zusätzliche Zähler- und Schalterarten

Dokumentation siehe: https://github.com/camueller/SmartApplianceEnabler/tree/2.0

**Comment by woupi** [comment]
Vielen Dank für die Info, damit werde ich mich dann, wenn es mein Zeitplan es zulässt, auf die 2 umstellen.

**Comment by camueller** [comment]
In diesem Fall bitte an die [Manuelle Installation für 2.0](https://github.com/camueller/SmartApplianceEnabler/blob/2.0/doc/ManualInstallation_DE.md) halten.

**Comment by woupi** [comment]
Ist bei mir sowieso nicht anders möglich, da ich keine Raspberry habe, sondern das ganze auf einem NUC und mit Proxmox läuft.
Darum muss ich auch ganz viel Zeit und Ruhe haben, möchte mir nichts zerschießen, nur weil ich zu schnell elementare Dinge vergessen habe und das ganze zwischen Tür und Angel versucht habe.
Ich Denke Weihnachten ist ein guter Zeitpunkt, da kann ich am Haus sonst eh nichts machen.
Ich werde dann gleich versuchen auch eine Anleitung zu erstellen, eventuell hat ja jetzt oder in der Zukunft eine ähnliche Konstellation.

**Reply by stevoboy** [reply]
Funktionieren dann nur noch mqtt basierte Zähler und Schalter, oder auch weiterhin http-basiert (get, Post etc.?)

**Reply by camueller** [reply]
Es funktioniert alles, was auch bisher funktioniert. Aber intern arbeitet der SAE jetzt mit MQTT an einigen Stellen, was die Visualiserung durch NodeRED erlaubt. Natürlich können diese MQTT-Nachrichten auch durch andere Systeme ausgelesen/verarbeitet werden.

**Comment by woupi** [comment]
Das ist eine gute Frage
